File: native-library/dothejob.c
char * doTheJobNative() {
    return "it works";
}

int doTheJobArrayNative(unsigned char inputArray[], int outputArray[], int arraySize) {
    for(int i=0; i < arraySize; i++) {
        outputArray[i]= inputArray[i]*2;
    }
    return arraySize;
}

File: src/main/java/org/tof/example/DoTheJob.java
package org.tof.example;

public class DoTheJob {
    static {
        new NativeLibraryLoader().loadNativeLibrarySafely("dothejob");
    }

    public native String doTheJobNative();

    public native int doTheJobArrayNative(byte[] inputArray, int[] outputArray, int arraySize);
}
